### PR TITLE
Fix app freeze when gridsteps are much larger than the actual plot range

### PR DESCRIPTION
**Pull request by Jonathan727**
Hello, firstly, sorry that this pull set has so many whitespace/formatting change. If you want me to submit a simplified pull request I can, just let me know. Android studio did that. The real important change is in XYGraphWidget.java#drawGrid(Canvas canvas).

I had been struggling for a while with an empty plot would freeze my app. I figured out that when the plot was empty, the domain grid loop was iterating one loop over a hundred million times.

I've rewritten the for loop and 3 others that are similar. I was very verbose in my commits so anyone can verify that the changes I made leave a loop that functions the same but in some cases skips a great many iterations that were no-op anyway. My reductions allowed me to remove the if condition inside the for loop.

### CODE REVIEW

**Comment by Jonathan727** [comment]
Sorry, I should have described how the long running loop actually froze my app.

Basically, the render thread would be calculating grids, iterating over 100 million times. Meanwhile, the main thread would be blocked at `synchronize(pingPong)`

**Comment by Jonathan727** [comment]
I made the grid loops much more compact and the logic easier to follow in commit 
02cdc94

**Comment by halfhp** [comment]
Thanks for the contribution @Jonathan727 !  I'll do my best to do a code review this evening.

File: androidplot-core/src/main/java/com/androidplot/Plot.java [hunk 1]
```diff
@@ -406,7 +421,7 @@ public void run() {
                     }
                     pingPong.recycle();
                 }
-            });
+            }, "Android Plot Render Thread");
```
Comment (halfhp) [inline androidplot-core/src/main/java/com/androidplot/Plot.java]: Would prefer `Androidplot` as one word.

**Comment by halfhp** [comment]
@Jonathan727 thanks for fixing!   I think this looks good.  For fixes and features I squash the changeset into a single commit so its easy to reference in the future.  If you prefer to squash the commit log yourself, feel free otherwise I'm more than happy to do it with the following message: (you'll still be listed as the author of the changeset)

> Fix app freeze when gridsteps are much larger than the actual plot range

**Comment by Jonathan727** [comment]
Hi @halfhp ,
I wondered if you might be willing to create a formal release on gradle with these changes. I think 1.5.4 is the last release and it doesn't include this fix. Not a huge priority since I know you are donating your time.

**Comment by halfhp** [comment]
Hey @Jonathan727 apologies for the delay.  I have one other issue to resolve and then should be ready to push a new release.  I'll do my best to get this done by Wednesday.

**Comment by halfhp** [comment]
@Jonathan727 it's now live in 1.5.5

**Comment by Jonathan727** [comment]
Thanks a Million!

-Jonathan (via mobile)

**Pull request merged**